application does not start when endpoint class in default package

### Describe the bug

when the endpoint class is located in the default package, the application fails to start.
this is new in `3.31.0.CR1`. this is working fine in `3.30.6`

### Expected behavior

application should start. default package, while not recommended, should be supported.

### Actual behavior

it fails with:
```
17:30:09,067 ERROR [io.qua.run.Quarkus] Error running Quarkus: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
        at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:501)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:78)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:50)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:143)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:69)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:37)
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 8 more
Caused by: java.lang.NoClassDefFoundError: GreetingResource_Bean
        at io.quarkus.arc.setup.Default_ComponentsProvider_addBeans0.addBeans0(Unknown Source)
        at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Unknown Source)
        at io.quarkus.arc.impl.ArcContainerImpl.<init>(ArcContainerImpl.java:137)
        at io.quarkus.arc.Arc.initialize(Arc.java:53)
        at io.quarkus.arc.runtime.ArcRecorder.initContainer(ArcRecorder.java:53)
        at io.quarkus.runner.recorded.ArcProcessor$initializeContainer643029769.deploy_0(Unknown Source)
        at io.quarkus.runner.recorded.ArcProcessor$initializeContainer643029769.deploy(Unknown Source)
        ... 9 more
Caused by: java.lang.ClassNotFoundException: GreetingResource_Bean
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
        at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:136)
        at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:87)
        ... 16 more
```

### How to Reproduce?

go to https://github.com/vsevel/quarkus_reproducers/tree/quarkus_3.31.0.CR1_default_package_endpoint and follow instructions in readme.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

3.31.0.CR1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/SerializedApplication.java
+++ b/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/SerializedApplication.java
@@ -250,9 +250,14 @@ public class SerializedApplication {
                 ZipEntry entry = entries.nextElement();
                 if (!entry.getName().contains("/")) {
                     hasDefaultPackage = true;
-                    if (!entry.getName().isEmpty() && FULLY_INDEXED_PATHS.contains("")) {
-                        fullyIndexedPaths.computeIfAbsent("", SerializedApplication::newFullyIndexedPathsValue)
-                                .add(entry.getName());
+                    if (!entry.getName().isEmpty()) {
+                        if (FULLY_INDEXED_PATHS.contains("")) {
+                            fullyIndexedPaths.computeIfAbsent("", SerializedApplication::newFullyIndexedPathsValue)
+                                    .add(entry.getName());
+                        }
+                        if (writeAllEntries) {
+                            allEntries.add(entry.getName());
+                        }
                     }
                 } else if (!entry.isDirectory()) {
                     //some jars don't have correct directory entries

